extmod/modssl_mbedtls.c

```diff
@@ -167,6 +246,9 @@ STATIC mp_obj_t ssl_context_make_new(const mp_obj_type_t *type_in, size_t n_args
     // Debug level (0-4) 1=warning, 2=info, 3=debug, 4=verbose
     mbedtls_debug_set_threshold(3);
     #endif
+    #ifdef MICROPY_MBEDTLS_PLATFORM_TIME_ALT
+    mbedtls_platform_set_time(platform_mbedtls_time);
```

Comment:
I suggest putting this function call in `esp32/main.c`, because it's very specific to the esp32 implementation.  All other ports use `MBEDTLS_PLATFORM_TIME_MACRO`, which is a simpler way to configure the time function.

(And if/when esp32 switches to 1970 as the Epoch, we can just delete this bit.)